images/router/haproxy/conf/haproxy-config.template

```diff
@@ -219,6 +219,7 @@ frontend fe_sni
     {{- if isTrue (env "ROUTER_STRICT_SNI") }} strict-sni {{ end }}
     {{- ""}} crt {{firstMatch ".+" .DefaultCertificate "/var/lib/haproxy/conf/default_pub_keys.pem"}}
     {{- ""}} crt-list /var/lib/haproxy/conf/cert_config.map accept-proxy
+    {{- if isTrue (env "ROUTER_ENABLE_HTTP2") }} alpn h2,http/1.1{{ end }}
```

Comment:
Do we really need ROUTER_ENABLE_HTTP2 env when we know that protocol downgrade happen automatically based on the client?

Reply:
It is intentionally disabled by default as per the requirements in the trello card: https://trello.com/c/qzvlzuyx/27-3-implement-router-http-2-support-terminating-at-the-router-router

Reply:
What is the underlying justification of the cited requirement for a new piece of config? What are the specific risks of always enabling http2? When would a user want to disable it?

Reply:
So as re: your questions:
3. The flag is flipped ... in that it is disabled by default and has to be explicitly enabled by the user/admin. 
2. I don't see risks to always enable http2 as we do downgrade the connections to http/1.x if http/2 is not supported but that said one reason I can see is this is a new feature (and so interactions with different/older clients are something to also consider) - making it more prudent to have a flag that enables it as needed.
1. @knobunc can probably more on the cited requirement. Suspect it might be 2 but ...

Reply:
Yeah, it's new in haproxy and there have already been security vulnerabilities around it.  I want to let it soak for a while before turning it on all the time.